message.go

```diff
@@ -59,6 +59,29 @@ func MarshalMessage(m Message) ([]byte, error) {
 	return im.Marshal(nil)
 }
 
+// MarshalMessageChecksum marshals a Message into its binary form and prepends
+// an ICMPv6 message with the correct type.
+func MarshalMessageChecksum(m Message, source, destination net.IP) ([]byte, error) {
+	mb, err := m.marshal()
```

Comment:
Let's factor out the body here into `func marshalMessage(m Message, psh []byte) ([]byte, error)` and then call it from both MarshalMessage (with psh nil) and MarshalMessageChecksum (with psh computed).